### PR TITLE
Hacktoberfest  - convert newline caracter to formart unix or dos

### PR DESCRIPTION
**Pull request by vicenthy**

### CODE REVIEW

**Comment by vicenthy** [comment]
Anyone can i help me with chacklist style plugin?

**Comment by grzesiekkedzior** [comment]
At first glance the Readme file has bad indentation. You are not using a local variable 'var' consistently. Some files have EOF. You try add snippet and on the other hand change another in one PR. ConvertNewLineFileToDosOrUnix.java file doesn't have License.

**Comment by vicenthy** [comment]
> At first glance the Readme file has bad indentation. You are not using a local variable 'var' consistently. Some files have EOF. You try add snippet and on the other hand change another in one PR. ConvertNewLineFileToDosOrUnix.java file doesn't have License.

tranks for your help :)

File: README.md [hunk 1]
```diff
@@ -42,7 +42,8 @@ For new snippets the general implementation steps are:
...
 [![Zip file](https://img.shields.io/badge/-Zip%20file-e1b050)](#zip-file) [![link](https://img.shields.io/badge/-Repository%20link-969c56?logo=github)](https://github.com/iluwatar/30-seconds-of-java/blob/master/src/main/java/file/ZipFileSnippet.java)  
-[![Zip multiple files](https://img.shields.io/badge/-Zip%20multiple%20files-e1b050)](#zip-multiple-files) [![link](https://img.shields.io/badge/-Repository%20link-969c56?logo=github)](https://github.com/iluwatar/30-seconds-of-java/blob/master/src/main/java/file/ZipFilesSnippet.java)
+[![Zip multiple files](https://img.shields.io/badge/-Zip%20multiple%20files-e1b050)](#zip-multiple-files) [![link](https://img.shields.io/badge/-Repository%20link-969c56?logo=github)](https://github.com/iluwatar/30-seconds-of-java/blob/master/src/main/java/file/ZipFilesSnippet.java)   
+[![Convert newline caracter unix to dos](https://img.shields.io/badge/-Convert%20newline%20caracter%20unix%20to%20dos-e1b050)](#Convert-newline-caracter-unix-to-dos) [![link](https://img.shields.io/badge/-Repository%20link-969c56?logo=github)](https://github.com/iluwatar/30-seconds-of-java/blob/master/src/main/java/io/ConvertNewLineFileToDosOrUnix.java)
```
Comment (iluwatar) [inline README.md:46]: Fix spelling, caracter -> character, unix -> Unix, dos -> DOS

File: src/main/java/file/ConvertNewLineFileToDosOrUnix.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+
+public class ConvertNewLineFileToDosOrUnix {
+  /**
+   * Convert Unix file LF to Windows File CRLF.
+   *
+   * @param fileIn  Path file to read and convert
+   * @param fileOut Path file to convert
```
Comment (iluwatar) [inline src/main/java/file/ConvertNewLineFileToDosOrUnix.java:40]: Should be output file

File: src/main/java/file/ConvertNewLineFileToDosOrUnix.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+   * @throws IOException if an I/O error occurs
+   */
+  public static void convertLfToCrlf(String fileIn, String fileOut, String charSet) 
+      throws IOException {
+    charSet = charSet != null ? charSet : "ISO-8859-1";
```
Comment (iluwatar) [inline src/main/java/file/ConvertNewLineFileToDosOrUnix.java:45]: Maybe check for empty string as well?

File: src/main/java/file/ConvertNewLineFileToDosOrUnix.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    BufferedWriter writer = new BufferedWriter(
+        new OutputStreamWriter(new FileOutputStream(new File(fileOut)), charSet));
+
+    BufferedReader reader = new BufferedReader(new FileReader(new File(fileIn)));
```
Comment (iluwatar) [inline src/main/java/file/ConvertNewLineFileToDosOrUnix.java:49]: I suggest we use try-with-resources to automatically close the writer and reader

File: src/main/java/file/ConvertNewLineFileToDosOrUnix.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    reader.lines().forEach(linha -> {
+      try {
+        writer.write(linha.replaceAll("\r", "") + "\r\n");
+      } catch (IOException e) {
+        e.printStackTrace();
```
Comment (iluwatar) [inline src/main/java/file/ConvertNewLineFileToDosOrUnix.java:54]: Probably would be better to let the exception be thrown to the caller

File: src/main/java/file/ConvertNewLineFileToDosOrUnix.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+  /**
+   * Convert Unix file CRLF to Windows File LF.
+   *
+   * @param fileIn  Path file to read and convert
+   * @param fileOut Path file to convert
+   * @throws IOException if an I/O error occurs
+   */
+  public static void convertCrlfToLf(String fileIn, String fileOut, String charSet)
+      throws IOException {
```
Comment (iluwatar) [inline src/main/java/file/ConvertNewLineFileToDosOrUnix.java:72]: Same comments as for the above method

File: src/main/java/string/ReversStringSnippet.java [hunk 7]
```diff
@@ -36,6 +38,7 @@
    * @return reversed string
    */
   public static String reverseString(String s) {
-    return new StringBuilder(s).reverse().toString();
+    String  result  = new String(s.getBytes(), StandardCharsets.UTF_8);
+    return new StringBuilder(result).reverse().toString();
```
Comment (iluwatar) [inline src/main/java/string/ReversStringSnippet.java:42]: Why this has been changed?

File: src/test/java/file/ConvertNewLineFileToDosOrUnixTest.java [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+public class ConvertNewLineFileToDosOrUnixTest {
+
+  @Test
+  public void testconvertLfToCrlf() {
+    try {
+      ConvertNewLineFileToDosOrUnix.convertLfToCrlf("src/test/resources/zlinuxFileText.txt",
+          "src/test/resources/windowsFileText.txt", null);
+      var file = new File("src/test/resources/windowsFileText.txt");
+      var totalLine =
+          Files.readAllLines(Paths.get("src/test/resources/windowsFileText.txt")).stream().count();
```
Comment (iluwatar) [inline src/test/java/file/ConvertNewLineFileToDosOrUnixTest.java:47]:
```suggestion
          Files.readAllLines(Paths.get("src/test/resources/windowsFileText.txt")).size();

```

File: src/test/java/file/ConvertNewLineFileToDosOrUnixTest.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        int count = 0;
+        while ((content = fr.read()) != -1) {
+          char c = (char) content;
+          if (c == '\n' | c == '\r') {
+            count++;
+          }
+        }
+        assertEquals(totalLine * 2, count);
+      } catch (IOException e) {
+        e.printStackTrace();
```
Comment (iluwatar) [inline src/test/java/file/ConvertNewLineFileToDosOrUnixTest.java:59]: Let the exception fly

File: src/test/java/file/ConvertNewLineFileToDosOrUnixTest.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+      }
+
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  @Test
+  public void testconvertCrlfToLf() {
```
Comment (iluwatar) [inline src/test/java/file/ConvertNewLineFileToDosOrUnixTest.java:68]: Same comments as for the above method

File: src/test/java/string/ReversStringSnippetTest.java [hunk 11]
```diff
@@ -39,6 +40,7 @@
   @Test
   void testReverseString() {
     assertEquals("oof", ReversStringSnippet.reverseString("foo"));
-    assertEquals("ÖÄÅ321FED cba", ReversStringSnippet.reverseString("abc DEF123ÅÄÖ"));
+    String normalizestring = new String("ÖÄÅ321FED cba".getBytes(), StandardCharsets.UTF_8);
+    assertEquals(normalizestring, ReversStringSnippet.reverseString("abc DEF123ÅÄÖ"));
```
Comment (iluwatar) [inline src/test/java/string/ReversStringSnippetTest.java:44]: No need for this change

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Comment by iluwatar** [comment]
@vicenthy please find the review comments above. Let me know when you're ready for another review by commenting in this thread.

**Review by PraveenNanda124 — COMMENTED** [review]
Good work

**Pull request closed without merging**